docs/device-api.md

```diff
@@ -0,0 +1,124 @@
+| `Logger`                  | level                 |    O     |       O        |      O       |     O      |      O       |
+|                           | log_path              |    O     |       O        |      O       |     O      |      O       |
+|                           | show_level            |    O     |       O        |      O       |     O      |      O       |
+|                           | show_log_origin       |    O     |       O        |      O       |     O      |      O       |
+| `MachineConfiguration`    | cpu_template          |    O     |       O        |      O       |     O      |      O       |
+|                           | ht_enabled            |    O     |       O        |      O       |     O      |      O       |
+|                           | mem_size_mib          |    O     |       O        |      O       |     O      |      O       |
+|                           | track_dirty_pages     |    O     |       O        |      O       |     O      |      O       |
+|                           | vcpu_count            |    O     |       O        |      O       |     O      |      O       |
+| `Metrics`                 | metrics_path          |    O     |       O        |      O       |     O      |      O       |
+| `MmdsConfig`              | ipv4_address          |    O     |       O        |      O       |   **R**    |      O       |
+| `NetworkInterface`        | allow_mmds_requests   |    O     |       O        |      O       |   **R**    |      O       |
+|                           | guest_mac             |    O     |       O        |      O       |   **R**    |      O       |
+|                           | host_dev_name         |    O     |       O        |      O       |   **R**    |      O       |
+|                           | iface_id              |    O     |       O        |      O       |   **R**    |      O       |
+|                           | rx_rate_limiter       |    O     |       O        |      O       |   **R**    |      O       |
+|                           | tx_rate_limiter       |    O     |       O        |      O       |   **R**    |      O       |
+| `PartialDrive`            | drive_id              |    O     |       O        |    **R**     |     O      |      O       |
+|                           | path_on_host          |    O     |       O        |    **R**     |     O      |      O       |
+| `PartialNetworkInterface` | iface_id              |    O     |       O        |      O       |   **R**    |      O       |
+|                           | rx_rate_limiter       |    O     |       O        |      O       |   **R**    |      O       |
+|                           | tx_rate_limiter       |    O     |       O        |      O       |   **R**    |      O       |
+| `RateLimiter`             | bandwidth             |    O     |       O        |    **R**     |   **R**    |      O       |
```

Comment:
IMO this should not be placed in the matrix for 2 reasons:
* RateLimiter` is not an API request that can be issued on its own. It can be used in either Net or Drive specific API requests.
* Someone reading this matrix can think that if they want to use the rate limiter configuration they would need to enable both virtio net and virtio block drivers which is not true.
My proposal is to remove this from here.

Reply:
> RateLimiter` is not an API request that can be issued on its own. It can be used in either Net or Drive specific API requests

Perhaps I misunderstand your objection? However the whole point of this table is to show which devices are required for and API request to 'work'.  Removing these from the table would suggest they can be expected to work regardless of the devices present.

Given the RateLimiter API should 'work with either device, a better approach would be to repeat the entry.  One row showing net device is required and block device optional.
The other entry showing the block device is required and the net device optional.

Read together, they both suggest that API will work if both devices are present.

Are their any other such API calls?

Reply:
As you said, "the whole point of this table is to show which devices are required for an API request to work" but RateLimiter does not have an API endpoint, it can't be issued as a standalone API request. Am I mistaken?

Reply:
> ... it can't be issued as a standalone API request.

But it is a API request?

Reply:
I can now understand better your view on the scope of matrix, I guess I understood its scope differently.  We can leave this as it is.